cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -459,6 +471,17 @@ public Void call() throws Exception {
           createStreamFeeds(config);
           alterExploreStream(streamId, true, config.getFormat());
           streamMetaStore.addStream(streamId, description);
+          // If an owner was provided then store it in owner store.
+          try {
+            if (!Strings.isNullOrEmpty(ownerPrincipal)) {
+              ownerStore.add(streamId, ownerPrincipal);
+            }
+          } catch (Exception e) {
+            // clean up from streamMetaStore
+            streamMetaStore.removeStream(streamId);
```

Comment:
I think that's a bigger conversation since the operations are not atomic. streamMetaStore.addStream(streamId, description); itself can fail with files left over. We should do it an independent PR if needed.